### PR TITLE
PWN-5210 - main page changes + Buy token logic changes + Banner changes

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5210

## Description of Work

PWN-5210 - buy tokens and changes on Main page and BuyInfoDetailsBottomSheet 

## Screenshot

Android Emulator - Pixel_4a_API_33_Ryan_:5554 2022-10-17 17-56-20

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 1]
```diff
@@ -51,11 +48,18 @@ class HomePresenter(
...
-    private val authInteractor: AuthInteractor
+    private val newBuyFeatureToggle: NewBuyFeatureToggle
 ) : BasePresenter<HomeContract.View>(), HomeContract.Presenter {
 
+    private lateinit var fallBackUsdcTokenForBuy: Token
+    private var nickName: Username? = null
+
+    init {
+        launch {
+            fallBackUsdcTokenForBuy = userInteractor.getTokensForBuy(listOf(USDC_SYMBOL)).first()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:59]: fallBack -> fallback

and i suggest you put this preloading on `onAttach` method and wrapping it in try catch

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 2]
```diff
@@ -51,11 +48,18 @@ class HomePresenter(
...
     private val homeElementItemMapper: HomeElementItemMapper,
     private val resourcesProvider: ResourcesProvider,
-    private val newBuyFeatureToggle: NewBuyFeatureToggle,
-    private val accountStorageContract: UserSignUpDetailsStorage,
-    private val authInteractor: AuthInteractor
+    private val newBuyFeatureToggle: NewBuyFeatureToggle
 ) : BasePresenter<HomeContract.View>(), HomeContract.Presenter {
 
+    private lateinit var fallBackUsdcTokenForBuy: Token
+    private var nickName: Username? = null
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:55]: nickName  -> username

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt [hunk 3]
```diff
@@ -107,10 +112,25 @@ class HomePresenter(
...
 
     override fun onBuyTokenClicked(token: Token) {
+        if (token.tokenSymbol !in TOKENS_VALID_FOR_BUY) {
+            view?.showBuyInfoScreen(token)
+        } else {
+            onBuyToken(token)
+        }
+    }
+
+    override fun onInfoBuyTokenClicked(token: Token) = onBuyToken(token)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:122]: block body

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:55]: I do not like userName.userName

File: app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt (hunk 3 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt:122]: :/ we need to think about not use block when case was just redirected ... 3 lines except of 1 line... wtf ?

**Review by 1sakov — APPROVED** [review]

**Pull request merged**